saleor/static/dashboard-next/categories/components/CategoryUpdatePage/CategoryUpdatePage.tsx

```diff
@@ -0,0 +1,289 @@
+import SeoForm from "../../../components/SeoForm";
+import Form from "../../../components/Form";
+import CategoryProductsCard from "../CategoryProductsCard";
+import ActionDialog from "../../../components/ActionDialog";
+import { MoneyType } from "../../../products";
+import Toggle from "../../../components/Toggle";
+import SaveButtonBar, {
+  SaveButtonBarState
+} from "../../../components/SaveButtonBar/SaveButtonBar";
+import Tabs, { SingleTab } from "../../../components/Tab";
+import { UserError } from "../../../";
+
+import i18n from "../../../i18n";
+import { CardSpacer } from "../../../components/CardSpacer";
+
+interface FormData {
+  description: string;
+  name: string;
+  seoTitle: string;
+  seoDescription: string;
+}
+
+export interface CategoryUpdatePageProps {
+  errors: UserError[];
+  disabled: boolean;
+  placeholderImage: string;
+  category: {
+    SeoDescription: string;
+    SeoTitle: string;
+    name: string;
+    description: string;
+  };
+  backgroundImage: {
+    url?: string;
+  };
+  products: Array<{
+    id: string;
+    name: string;
+    productType: {
+      name: string;
+    };
+    thumbnailUrl: string;
+    availability: {
+      available: boolean;
+    };
+    price: MoneyType;
+  }>;
+  subcategories: Array<{
+    id: string;
+    name: string;
+    children: {
+      totalCount: number;
+    };
+    products: {
+      totalCount: number;
+    };
+  }>;
+  pageInfo: {
+    hasNextPage: boolean;
+    hasPreviousPage: boolean;
+  };
+  loading: boolean;
+  saveButtonBarState?: SaveButtonBarState;
+  onSubmit(data: FormData);
+  onImageUpload?(event: React.ChangeEvent<any>);
+  onNextPage();
+  onPreviousPage();
+  onProductClick(id: string): () => void;
+  onAddProduct();
+  onBack();
+  onDelete();
+  onImageDelete: (id: string) => () => void;
+  onAddCategory();
+  onCategoryClick(id: string): () => void;
+}
+
+const decorate = withStyles(theme => ({
+  root: {
+    marginTop: theme.spacing.unit * 2,
+    gridGap: theme.spacing.unit * 4 + "px"
+  },
+  tabsBorder: {
+    borderBottom: "1px solid #eeeeee"
+  }
+}));
+
+export const CategoryUpdatePage = decorate<CategoryUpdatePageProps>(
+  ({
+    classes,
+    disabled,
+    errors: userErrors,
+    category,
+    onImageUpload,
+    subcategories,
+    loading,
+    products,
+    pageInfo,
+    onNextPage,
+    onPreviousPage,
+    onProductClick,
+    onAddProduct,
+    onDelete,
+    saveButtonBarState,
+    onSubmit,
+    onBack,
+    onAddCategory,
+    onCategoryClick,
+    onImageDelete,
+    placeholderImage,
+    backgroundImage
+  }) => {
+    const initialData = category
+      ? {
+          name: category.name,
+          description: category.description,
+          seoTitle: category.SeoTitle,
+          seoDescription: category.SeoDescription
+        }
+      : {
+          name: "",
+          description: "",
+          seoTitle: "",
+          seoDescription: ""
+        };
+    const isRoot = !loading && !products && !category;
```

Comment:
Category root view should be extracted to its own component to simplify component logic and evade skeleton mismatches